crates/oxc_linter/src/service.rs

```diff
@@ -236,13 +236,15 @@ impl Runtime {
             self.update_cache_state(path);
 
             // Retrieve all dependency modules from this module.
-            let dir = path.parent().unwrap();
+            // Since the path maybe a relative path, we need to canonicalize it before pass into
+            // resolver, related to #1887
+            let dir = dunce::canonicalize(path.parent().unwrap()).unwrap();
```

Comment:
I don't think is the correct way, we shouldn't hit the disk again after getting paths from the walker.

Reply:
if path is `../test_dir/something` and specifier is `something.js`, the resolver will return `test_dir/something.js`, but the correct path should be `../test_dir/something/something.js`. you could see https://github.com/oxc-project/oxc/issues/1887#issue-2065115177 as a example.

Reply:
Do you mean we should `canonicalize` the path at the beginning?

Reply:
Or fixed the issue in `oxc_resolver` ?